app/controllers/results_controller.rb

```diff
@@ -138,12 +138,11 @@ def edit
 
   def run_tests
     grouping_id = params[:grouping_id]
-    # TODO: The submission id is set incorrectly as the grouping_id
     submission_id = Result.find(params[:id]).submission.id
 
     begin
       AutomatedTestsHelper.request_a_test_run(grouping_id,
-                                              'submission',
+                                              'request',
```

Comment:
Making the button do the correct thing discovered some errors in the marking phase that happen only with multiple test runs. I'll make sure to fix that too before recreating this pull request with the changes we discussed.